src/views/HomeRedesign/HomeSection.tsx

```diff
@@ -0,0 +1,68 @@
+import { Heading } from "@chakra-ui/react";
+import { Fragment, FunctionComponent, useMemo } from "react";
+import { CatalogPackage } from "../../api/package/packages";
+import { PackageList } from "../../components/PackageList";
+import { useCatalog } from "../../contexts/Catalog";
+
+export interface HomeSectionProps {
+  name: string;
+  showLastUpdated?: number;
+  showPackages?: string[];
+}
+
+export const HomeSection: FunctionComponent<HomeSectionProps> = ({
+  name,
+  showLastUpdated,
+  showPackages,
+}) => {
+  const { data, loading, error } = useCatalog();
+
+  const results = useMemo(() => {
+    if (loading || error || !data?.packages) return [];
+
+    let packages = data.packages;
+
+    if (showLastUpdated) {
+      // sort packages by date
+      packages = [...packages]
+        .sort((p1, p2) => {
+          const d1 = new Date(p1.metadata.date);
+          const d2 = new Date(p2.metadata.date);
+          if (d1 === d2) {
+            return 0;
+          }
+          return d1 < d2 ? 1 : -1;
+        })
+        .slice(0, showLastUpdated);
+    } else if (showPackages) {
+      // find the specific packages from the catalog and
+      // return them in the same order
+      const filteredMap = [...packages].reduce(
```

Comment:
micro nitpick, but there's no need to copy `packages` here since `.reduce` doesn't mutate the original